src/server/src/classes/Action.ts

```diff
@@ -63,55 +50,72 @@ export class Action {
     return true;
   }
 
-  call(): boolean {
+  bet(): boolean {
+    const betAmount = this.action.betAmount as number;
+    const callAmount = this.round.highestBet - this.player.currentBet;
+
+    const mainCases = (): boolean => {
```

Comment:
it’s kinda confusing having this block of code up here like this - i think the bet() function would be clearer as a bunch of chained else ifs - but i’ll leave it to your discretion.